sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -2700,20 +2700,24 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
       name: TableIdentifier,
       nameParts: String): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1283",
+      errorClass = "INVALID_TEMP_OBJ_REFERENCE",
       messageParameters = Map(
-        "name" -> name.toString,
-        "nameParts" -> nameParts))
+        "obj" -> "view",
+        "objName" -> name.toString,
```

Comment:
Could you quote the table id by using `toSQLId`:
```suggestion
        "objName" -> toSQLId(name.nameParts),```